<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RuleEdit</class>
 <widget class="QWidget" name="RuleEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>581</width>
    <height>611</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>9</number>
   </property>
   <item>
    <widget class="QGroupBox" name="generalGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>General</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::FieldsStayAtSizeHint</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="nameLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Rule name:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="nameEdit"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Left-hand side</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Developer::GraphWidget" name="lhsGraph"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Right-hand side</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Developer::GraphWidget" name="rhsGraph"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="variablesLayout">
       <item>
        <layout class="QHBoxLayout" name="variablesLabelLayout">
         <item>
          <widget class="QLabel" name="variablesLabel">
           <property name="text">
            <string>Variables</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableWidget" name="variablesWidget">
         <property name="maximumSize">
          <size>
           <width>260</width>
           <height>16777215</height>
          </size>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string>New Row</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>New Row</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>New Row</string>
          </property>
         </row>
         <column>
          <property name="text">
           <string>Identifier</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Type</string>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string/>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="behaviourGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Condition</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="conditionsLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Only match when:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Developer::ConditionEditor" name="conditionsEdit"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Developer::GraphWidget</class>
   <extends>QGraphicsView</extends>
   <header>graphview/graphwidget.hpp</header>
  </customwidget>
  <customwidget>
   <class>Developer::ConditionEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>conditioneditor.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>nameEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>RuleEdit</receiver>
   <slot>nameChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>318</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>conditionsEdit</sender>
   <signal>textChanged()</signal>
   <receiver>RuleEdit</receiver>
   <slot>conditionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>508</x>
     <y>535</y>
    </hint>
    <hint type="destinationlabel">
     <x>607</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showInjectiveHelp()</slot>
  <slot>nameChanged(QString)</slot>
  <slot>documentationChanged()</slot>
  <slot>lhsChanged()</slot>
  <slot>rhsChanged()</slot>
  <slot>injectiveChanged(int)</slot>
  <slot>conditionChanged()</slot>
 </slots>
</ui>
